activation/handler.go

```diff
@@ -165,47 +162,84 @@ func (h *Handler) ProcessAtx(ctx context.Context, atx *types.VerifiedActivationT
 	return nil
 }
 
-// SyntacticallyValidateAtx ensures the following conditions apply, otherwise it returns an error.
-//
-//   - The PublishEpoch is less than or equal to the current epoch + 1.
-//   - If the sequence number is non-zero: PrevATX points to a syntactically valid ATX whose sequence number is one less
-//     than the current ATXs sequence number.
-//   - If the sequence number is zero: PrevATX is empty.
-//   - Positioning ATX points to a syntactically valid ATX.
-//   - NIPost challenge is a hash of the serialization of the following fields:
-//     NodeID, SequenceNumber, PrevATXID, LayerID, StartTick, PositioningATX.
-//   - The NIPost is valid.
-//   - ATX LayerID is NIPostLayerTime or less after the PositioningATX LayerID.
-//   - The ATX view of the previous epoch contains ActiveSetSize activations.
-func (h *Handler) SyntacticallyValidateAtx(ctx context.Context, atx *types.ActivationTx) (*types.VerifiedActivationTx, error) {
+func (h *Handler) SyntacticallyValidate(ctx context.Context, atx *types.ActivationTx) error {
+	if atx.NIPost == nil {
+		return fmt.Errorf("nil nipst for atx %s", atx.ShortString())
+	}
+	current := h.clock.CurrentLayer().GetEpoch()
+	if atx.PublishEpoch > current+1 {
+		return fmt.Errorf("atx publish epoch is too far in the future: %d > %d", atx.PublishEpoch, current+1)
+	}
+	if atx.PositioningATX == types.EmptyATXID {
+		return fmt.Errorf("empty positioning atx")
+	}
+
+	switch {
+	case atx.PrevATXID == types.EmptyATXID:
+		if atx.InitialPost == nil {
+			return fmt.Errorf("no prev atx declared, but initial post is not included")
+		}
+		if atx.InnerActivationTx.NodeID == nil {
+			return fmt.Errorf("no prev atx declared, but node id is missing")
+		}
+		if atx.VRFNonce == nil {
+			return fmt.Errorf("no prev atx declared, but vrf nonce is missing")
+		}
+		if atx.CommitmentATX == nil {
+			return fmt.Errorf("no prev atx declared, but commitment atx is missing")
+		}
+		if *atx.CommitmentATX == types.EmptyATXID {
+			return fmt.Errorf("empty commitment atx")
+		}
+		if atx.Sequence != 0 {
+			return fmt.Errorf("no prev atx declared, but sequence number not zero")
+		}
+
+		// Use the NIPost's Post metadata, while overriding the challenge to a zero challenge,
+		// as expected from the initial Post.
+		initialPostMetadata := *atx.NIPost.PostMetadata
+		initialPostMetadata.Challenge = shared.ZeroChallenge
+		if err := h.nipostValidator.Post(ctx, atx.SmesherID, *atx.CommitmentATX, atx.InitialPost, &initialPostMetadata, atx.NumUnits); err != nil {
+			return fmt.Errorf("invalid initial post: %w", err)
+		}
+		if err := h.nipostValidator.VRFNonce(atx.SmesherID, *atx.CommitmentATX, atx.VRFNonce, &initialPostMetadata, atx.NumUnits); err != nil {
+			return fmt.Errorf("invalid vrf nonce: %w", err)
+		}
```

Comment:
NIT: The order of these checks could be switched as verifying VRF nonce should be much faster.